packages/graphql-hooks/src/LocalGraphQLClient.js

```diff
@@ -0,0 +1,54 @@
+import GraphQLClient from './GraphQLClient'
+
+/** Local version of the GraphQLClient which only returns specified queries
+ * Meant to be used as a way to easily mock and test queries during development. This client never contacts any actual server.
+ * Queries are given in the form of an object of functions.
+ * Example:
+ ```
+    const localQueries = {
+      [allPostsQuery]: () => ({
+        allPosts: [
+          {
+            id: 1,
+            title: 'Test',
+            url: 'https://example.com'
+          }
+        ]
+      }),
+      [createPostMutation]: () => ({ createPost: { id: 1 } }),
+    }
+    const client = new LocalGraphQLClient({ localQueries })
+  ```
+ */
+class LocalGraphQLClient extends GraphQLClient {
```

Comment:
I wonder, do we really need to extend the client? Why do we need to inherit from it? Is there anything that the subclass needs from the base class?

Reply:
It's mostly to make it easily conform to the same interface and accept the same options (even if most will end up unused in the case of the local client). It could probably work even if we don't extend I suppose, but this way the relationship between the two is pretty clear

Reply:
Unfortunately the internal hooks like `useClientRequest` and `useQuery` [depends](https://github.com/nearform/graphql-hooks/blob/master/packages/graphql-hooks/src/useClientRequest.js#L159) on [many](https://github.com/nearform/graphql-hooks/blob/master/packages/graphql-hooks/src/useClientRequest.js#L201) [methods](https://github.com/nearform/graphql-hooks/blob/master/packages/graphql-hooks/src/useQuery.js#L82) from `GraphQLClient`.

By extending we are sure that we don't break any of these dependencies.

Would be nice to be able to do something like [urql](https://formidable.com/open-source/urql/docs/advanced/testing/#handling-multiple-hooks) where to create a client is enough to provide an object with the request methods.

Since the access to `GraphQLClient` and all it's methods is public, to get that we would have to introduce some breaking changes.

Reply:
That's why I mentioned it's just an API issue, not a functionality issue. The local client uses little to nothing of the real client, so an interface would do. On the other hand, I believe that turning it into an interface will be a breaking change, hence why I'm ok with inheritance